src/actions/actions.js

```diff
@@ -411,6 +412,126 @@ export function markCellNotRendered() {
   }
 }
 
+function loginSuccess(userData) {
+  return (dispatch) => {
+    dispatch({
+      type: 'LOGIN_SUCCESS',
+      userData,
+    })
+    dispatch(updateAppMessages('You are logged in'))
+  }
+}
+
+function loginFailure() {
+  return (dispatch) => {
+    dispatch(updateAppMessages('Login Failed'))
+  }
+}
+
+export function login() {
+  const url = '/auth/github'
+  const name = 'github_login'
+  const specs = 'width=500,height=600'
+  const authWindow = window.open(url, name, specs)
+  authWindow.focus()
+
+  return (dispatch) => {
+    // Functions to be called by child window
+    window.loginSuccess = (userData) => {
+      dispatch(loginSuccess(userData))
+    }
+    window.loginFailure = () => dispatch(loginFailure())
+  }
+}
+
+export function logout() {
+  return (dispatch) => {
+    fetch('/logout')
+      .then(response => response.json())
+      .then((json) => {
+        if (json.status === 'success') {
+          dispatch({ type: 'LOGOUT' })
+          dispatch(updateAppMessages('Logged Out'))
+        } else dispatch(updateAppMessages('Logout Failed'))
+      })
+  }
+}
+
+export function exportGist() {
+  // Go through all gist of the user
+  // Match the discription and fileName for each gist
+  // If mactch found, update it
+  // If none found then create a new Gist
+  const API_ROUTE = 'https://api.github.com'
+  return (dispatch, getState) => {
+    const state = getState()
+    const filename = titleToHtmlFilename(state.title)
+    let matchDescription
+    let gistExist = false
+    const gistData = {
+      description: state.title,
+      public: true,
+      files: {
+        [filename]: { content: exportJsmdBundle(state) },
+      },
+    };
+    fetch(`${API_ROUTE}/gists?access_token=${state.userData.accessToken}`)
+      .then(response => response.json())
+      .then((json) => {
+        matchDescription = json.filter(gist =>
+          gist.description === gistData.description &&
+          Object.keys(gist.files).length === 1 &&
+          Object.keys(gist.files)[0] === filename)
+
+        if (!matchDescription.length) {
+          return fetch(`${API_ROUTE}/gists?access_token=${state.userData.accessToken}`, {
+            body: JSON.stringify(gistData),
+            method: 'POST',
+          })
+        }
+        gistExist = true
+        const gistID = matchDescription[0].id
+        return fetch(`${API_ROUTE}/gists/${gistID}?access_token=${state.userData.accessToken}`, {
+          body: JSON.stringify(gistData),
+          method: 'PATCH',
+        })
+      })
+      .then(response => response.json())
+      .then((json) => {
+        const message = gistExist ? 'Updated Gist' : 'Exported to GitHub Gist'
+        dispatch(updateAppMessages(`${message}:
+<a href="${json.html_url}" target="_blank">Gist</a> -
+<a href="https://iodide-project.github.io/master/?gist=${json.owner.login}/${json.id}" target="_blank"> Runnable notebook</a>`))
```

Comment:
Does this do anything yet? Not seeing anything on the iodide-sv prototype

Reply:
@wlach it will once #617 lands, though this will have to change to reflect the knew schema.

Reply:
Oh, and also to be clear, this currently adds something to the App Messages pane. Once we have the snackbar, then we'll get a small elem that pops up on bottom of the screen.